### DISCUSSION TITLE
How to use the extension if functions return instance of "Redis" instead of data?

### DISCUSSION THREAD

**Discussion by tomislavcivcija** [phpredis/phpredis#2447] [Q&A] [answered]
As i see in the official stubs 
https://github.com/phpredis/phpredis/blob/develop/redis.stub.php

almost every function returns what it is supposed to, OR it returns "Redis". What? How do we interpret that result? Please give me example, in the whole README of the project i was not able to find a mention of this. Even the docs seem wrong, for sMembers (and many others) they say
```
Return value
An array of elements, the contents of the set.
```
"Redis" is not mentioned.

Why i ask this? It broke my prod, because sMembers returned "Redis" instead of an array, now i want to fix the code but i have no idea how to go about it. The thing is, the redis connection works (it is a long-running 7 months cli php script that uses this single connection), everything works (get/set) - its just the sMembers function that started returning "Redis" after these 7 months.

So, how should i interpret when functions return "Redis" instead of actual data, do i interpret it like a connection gone bad (even though the connection actually works fine..)?

**Comment by yatsukhnenko** [comment, marked as answer]
PhpRedis returns Redis object when it is in multi/pipeline mode. Here is an example how to use it
https://github.com/phpredis/phpredis?tab=readme-ov-file#example-115

**Reply by tomislavcivcija** [reply]
Wow okay got it now, it was so damn confusing looking at the logs and then seeing stubs with operations returning Redis.

I used RSMQ library, took a look at its source code and sometimes it uses multi(), most likely it didnt handle an exception properly and it left the redis connection in multi mode.. thanks for your answer
